core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -607,4 +610,17 @@ class AdminManager(val config: KafkaConfig,
     val readOnly = !allNames.exists(DynamicBrokerConfig.AllDynamicConfigs.contains)
     new DescribeConfigsResponse.ConfigEntry(name, valueAsString, source, isSensitive, readOnly, synonyms.asJava)
   }
+
+  private def resolveVariableConfigs(propsOriginal: Properties): Properties = {
+    val props = new Properties
+    propsOriginal.put("bootstrap.servers", "localhost:1")
```

Comment:
What is the purpose of this `put`? As far as I can tell, it won't get added to the `props` object anyway, as we filter on the key below

Reply:
I reverted changes because this solution will write resolved config to ZK.